sdk/eventgrid/Azure.Messaging.EventGrid/README.md

```diff
@@ -197,24 +197,37 @@ Below is an example calling `GetData()` using Event Grid events. Calling `GetDat
 ```C# Snippet:DeserializePayloadUsingNonGenericGetData
 foreach (EventGridEvent egEvent in egEvents)
 {
-    // If the event is a system event, GetData() should return the correct system event type
-    switch (egEvent.GetData())
+    // If the event is a system event, AsSystemEventData() should return the correct system event type
```

Comment:
Is the prose above in the README.md still correct?  I think we are making `GetData()` now always return `BinaryData`?

Reply:
GetData is still used in the else case, but I agree with the change to always return BinaryData, the sample should probably be updated as it doesn't demonstrate a good pattern.

Reply:
I think maybe I should have been more explicit.  If you expand the hunk here up (it's not included in the default view that GitHub shows), you see we have this text:

> Below is an example calling `GetData()` using Event Grid events. Calling `GetData()` will either return a deserialized **system event** (an event generated by an Azure service), or an object of type `BinaryData`, which represents the serialized JSON event data as bytes.

With your changes, I think that `GetData()` now always returns a `BinaryData`?

Reply:
Ah yes, that too 😄